lldb/source/Target/Target.cpp

```diff
@@ -1438,6 +1439,54 @@ static void LoadScriptingResourceForModule(const ModuleSP &module_sp,
                                                   feedback_stream.GetData());
 }
 
+static void LoadTypeSummariesForModule(ModuleSP module_sp) {
+  auto *sections = module_sp->GetSectionList();
+  if (!sections)
+    return;
+
+  llvm::Triple triple = module_sp->GetArchitecture().GetTriple();
+  SectionSP summaries_sp;
+  if (triple.isOSDarwin()) {
+    if (auto data_sp = sections->FindSectionByName(ConstString("__DATA_CONST")))
+      summaries_sp = data_sp->GetChildren().FindSectionByName(
+          ConstString("__lldbsummaries"));
+  } else if (triple.isOSLinux() || triple.isOSWindows()) {
+    summaries_sp = sections->FindSectionByName(ConstString(".lldbsummaries"));
```

Comment:
This knowledge about the exact section name should be abstracted behind a virtual method in ObjectFile, which (at least in the LLVM variant) already has a mechanism to access differently named sections.

Reply:
Given that this is the only place the names are used, I wonder if it's worth adding abstraction, worth expanding the `ObjectFile` interface?

Reply:
There's also `SwiftObjectFileFormat` at `swift/include/swift/ABI/ObjectFile.h`

Reply:
Instead of putting forward a question, I will say: I think it's there's value in not adding abstraction in this case. My arguments against adding an abstraction are:
* These names aren't used anywhere else, and I don't see that changing
* Spreads the knowledge across multiple files vs having the names in one place
* Expands the API of `ObjectFile` etc

If in the future these names do get used from multiple locations, the abstraction can be added at that time.

Reply:
What do you think about adding cases to FindSectionByType? 
https://lldb.llvm.org/cpp_reference/classlldb__private_1_1SectionList.html#ad8f56673377eb9d1fc130724d1ef6a4f

Reply:
https://lldb.llvm.org/cpp_reference/namespacelldb.html#ad269ed92bcd3705de1abb636f0a5fa1c

Reply:
I like that, thanks.

Reply:
See 137a2116a0907f26467ef6afb253fc6b63eaee36